{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Here we get gene information, but unsure exactly what this is. It's not expression data. Looks like the sets of genes associated with\n",
    "# one purturbation\n",
    "clue_gene = pd.read_csv('/home/jovyan/Tomics-CP-Chem-MoA/04_Tomics_Models/init_data_expl/clue_geneinfo_beta.txt', delimiter = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_title</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>src</th>\n",
       "      <th>feature_space</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>750</td>\n",
       "      <td>GAS8-AS1</td>\n",
       "      <td>ENSG00000221819</td>\n",
       "      <td>GAS8 antisense RNA 1</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6315</td>\n",
       "      <td>ATXN8OS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATXN8 opposite strand lncRNA</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7503</td>\n",
       "      <td>XIST</td>\n",
       "      <td>ENSG00000229807</td>\n",
       "      <td>X inactive specific transcript</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>8552</td>\n",
       "      <td>INE1</td>\n",
       "      <td>ENSG00000224975</td>\n",
       "      <td>inactivation escape 1</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9834</td>\n",
       "      <td>FAM30A</td>\n",
       "      <td>ENSG00000226777</td>\n",
       "      <td>family with sequence similarity 30 member A</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10141</td>\n",
       "      <td>LINC01587</td>\n",
       "      <td>ENSG00000082929</td>\n",
       "      <td>long intergenic non-protein coding RNA 1587</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10255</td>\n",
       "      <td>HCG9</td>\n",
       "      <td>ENSG00000204625</td>\n",
       "      <td>HLA complex group 9</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10281</td>\n",
       "      <td>DSCR4</td>\n",
       "      <td>ENSG00000184029</td>\n",
       "      <td>Down syndrome critical region 4</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>23434</td>\n",
       "      <td>LINC01565</td>\n",
       "      <td>ENSG00000198685</td>\n",
       "      <td>long intergenic non-protein coding RNA 1565</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>23591</td>\n",
       "      <td>FAM215A</td>\n",
       "      <td>ENSG00000267496</td>\n",
       "      <td>family with sequence similarity 215 member A</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>inferred</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene_id gene_symbol       ensembl_id  \\\n",
       "0      750    GAS8-AS1  ENSG00000221819   \n",
       "1     6315     ATXN8OS              NaN   \n",
       "2     7503        XIST  ENSG00000229807   \n",
       "3     8552        INE1  ENSG00000224975   \n",
       "4     9834      FAM30A  ENSG00000226777   \n",
       "5    10141   LINC01587  ENSG00000082929   \n",
       "6    10255        HCG9  ENSG00000204625   \n",
       "7    10281       DSCR4  ENSG00000184029   \n",
       "8    23434   LINC01565  ENSG00000198685   \n",
       "9    23591     FAM215A  ENSG00000267496   \n",
       "\n",
       "                                     gene_title gene_type   src feature_space  \n",
       "0                          GAS8 antisense RNA 1     ncRNA  NCBI      inferred  \n",
       "1                  ATXN8 opposite strand lncRNA     ncRNA  NCBI      inferred  \n",
       "2                X inactive specific transcript     ncRNA  NCBI      inferred  \n",
       "3                         inactivation escape 1     ncRNA  NCBI      inferred  \n",
       "4   family with sequence similarity 30 member A     ncRNA  NCBI      inferred  \n",
       "5   long intergenic non-protein coding RNA 1587     ncRNA  NCBI      inferred  \n",
       "6                           HLA complex group 9     ncRNA  NCBI      inferred  \n",
       "7               Down syndrome critical region 4     ncRNA  NCBI      inferred  \n",
       "8   long intergenic non-protein coding RNA 1565     ncRNA  NCBI      inferred  \n",
       "9  family with sequence similarity 215 member A     ncRNA  NCBI      inferred  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clue_gene[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene_id', 'gene_symbol', 'ensembl_id', 'gene_title', 'gene_type',\n",
       "       'src', 'feature_space'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clue_gene.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12328, 7)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clue_gene.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "clue_gene = clue_gene[clue_gene.feature_space == \"landmark\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>ensembl_id</th>\n",
       "      <th>gene_title</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>src</th>\n",
       "      <th>feature_space</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2154</th>\n",
       "      <td>16</td>\n",
       "      <td>AARS</td>\n",
       "      <td>ENSG00000090861</td>\n",
       "      <td>alanyl-tRNA synthetase</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2155</th>\n",
       "      <td>23</td>\n",
       "      <td>ABCF1</td>\n",
       "      <td>ENSG00000204574</td>\n",
       "      <td>ATP binding cassette subfamily F member 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>25</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>ENSG00000097007</td>\n",
       "      <td>ABL proto-oncogene 1, non-receptor tyrosine ki...</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2157</th>\n",
       "      <td>30</td>\n",
       "      <td>ACAA1</td>\n",
       "      <td>ENSG00000060971</td>\n",
       "      <td>acetyl-CoA acyltransferase 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2158</th>\n",
       "      <td>39</td>\n",
       "      <td>ACAT2</td>\n",
       "      <td>ENSG00000120437</td>\n",
       "      <td>acetyl-CoA acetyltransferase 2</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3127</th>\n",
       "      <td>200081</td>\n",
       "      <td>TXLNA</td>\n",
       "      <td>ENSG00000084652</td>\n",
       "      <td>taxilin alpha</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3128</th>\n",
       "      <td>200734</td>\n",
       "      <td>SPRED2</td>\n",
       "      <td>ENSG00000198369</td>\n",
       "      <td>sprouty related EVH1 domain containing 2</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3129</th>\n",
       "      <td>256364</td>\n",
       "      <td>EML3</td>\n",
       "      <td>ENSG00000149499</td>\n",
       "      <td>EMAP like 3</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3130</th>\n",
       "      <td>375346</td>\n",
       "      <td>STIMATE</td>\n",
       "      <td>ENSG00000213533</td>\n",
       "      <td>STIM activating enhancer</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3131</th>\n",
       "      <td>388650</td>\n",
       "      <td>DIPK1A</td>\n",
       "      <td>ENSG00000154511</td>\n",
       "      <td>divergent protein kinase domain 1A</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NCBI</td>\n",
       "      <td>landmark</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>978 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene_id gene_symbol       ensembl_id  \\\n",
       "2154       16        AARS  ENSG00000090861   \n",
       "2155       23       ABCF1  ENSG00000204574   \n",
       "2156       25        ABL1  ENSG00000097007   \n",
       "2157       30       ACAA1  ENSG00000060971   \n",
       "2158       39       ACAT2  ENSG00000120437   \n",
       "...       ...         ...              ...   \n",
       "3127   200081       TXLNA  ENSG00000084652   \n",
       "3128   200734      SPRED2  ENSG00000198369   \n",
       "3129   256364        EML3  ENSG00000149499   \n",
       "3130   375346     STIMATE  ENSG00000213533   \n",
       "3131   388650      DIPK1A  ENSG00000154511   \n",
       "\n",
       "                                             gene_title       gene_type   src  \\\n",
       "2154                             alanyl-tRNA synthetase  protein-coding  NCBI   \n",
       "2155          ATP binding cassette subfamily F member 1  protein-coding  NCBI   \n",
       "2156  ABL proto-oncogene 1, non-receptor tyrosine ki...  protein-coding  NCBI   \n",
       "2157                       acetyl-CoA acyltransferase 1  protein-coding  NCBI   \n",
       "2158                     acetyl-CoA acetyltransferase 2  protein-coding  NCBI   \n",
       "...                                                 ...             ...   ...   \n",
       "3127                                      taxilin alpha  protein-coding  NCBI   \n",
       "3128           sprouty related EVH1 domain containing 2  protein-coding  NCBI   \n",
       "3129                                        EMAP like 3  protein-coding  NCBI   \n",
       "3130                           STIM activating enhancer  protein-coding  NCBI   \n",
       "3131                 divergent protein kinase domain 1A  protein-coding  NCBI   \n",
       "\n",
       "     feature_space  \n",
       "2154      landmark  \n",
       "2155      landmark  \n",
       "2156      landmark  \n",
       "2157      landmark  \n",
       "2158      landmark  \n",
       "...            ...  \n",
       "3127      landmark  \n",
       "3128      landmark  \n",
       "3129      landmark  \n",
       "3130      landmark  \n",
       "3131      landmark  \n",
       "\n",
       "[978 rows x 7 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clue_gene"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Things to Try\n",
    "1. We could start by just looking at the genes from the transcriptomic profiles that are landmark genes, since the others are inferred from these genes anyway.\n",
    "2. Then use the transcriptomic profiles from the 10 MoAs with the most different type of compounds\n",
    "3. For a single compound, we could see how the average transcriptomic values for a single MoA ranges over different cell lines using the landmark genes.\n",
    "    - if there isn't a correlation, could mean a lot of noise due to the signal coming from the cell lines\n",
    "    - if there is a correlation, would support that we could use multiple cell lines to draw conclusions about the MoA\n",
    "4. Then implement an adapted version of the second place solution found on kaggle (maybe the first if I understand)\n",
    "    - remove cell viability data\n",
    "    - adapt input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
